src/inmanta/server/services/compilerservice.py

```diff
@@ -1068,3 +1068,18 @@ async def environment_action_deleted(self, env: model.Environment) -> None:
     async def recalculate_queue_count_cache(self) -> None:
         async with self._queue_count_cache_lock:
             self._queue_count_cache = await data.Compile.get_total_length_of_all_compile_queues()
+
+    async def cancel_compile(self, env_id: uuid.UUID) -> None:
+        """
+        Cancel the ongoing compile for the given environment (if any) and don't start any new compiles.
```

Comment:
> and don't start any new compiles

I don't think this is enforced by this method.

Reply:
I mean that this method doesn't start any new compiles. What do you propose?